hack/installers/install-codegen-go-tools.sh

```diff
@@ -9,10 +9,12 @@ SRCROOT="$( CDPATH='' cd -- "$(dirname "$0")/../.." && pwd -P )"
 # unecessary dependencies to go.mod). We want to maintain a single source of truth for versioning
 # our binaries (either go.mod or go install <pkg>@<version>), so we use two techniques to install
 # our CLIs:
-# 1. For CLIs which are NOT vendored in go.mod, we can run `go install` with an explicit version
-# 2. For packages which we *do* vendor in go.mod, we can run `go install` from the vendor directory
+# 1. For CLIs which are NOT vendored in go.mod, we can run `go install <pkg>@<version>` with an explicit version
+# 2. For packages which we *do* vendor in go.mod, we determine version from go.mod followed by `go install` with that version
 go_mod_install() {
-    go install -mod=vendor ./vendor/$1
+    module=$(go list -f '{{.Module}}' $1 | awk '{print $1}')
```

Comment:
@jessesuen Just tried to build the test-tools-image (via `make test-tools-image`) and still seeing related issues to having this executed in docker:
```sh
+++ dirname ./installers/install-codegen-go-tools.sh
++ CDPATH=
++ cd -- ./installers/../..
++ pwd -P
+ SRCROOT=/
+ export GOBIN=//dist
+ GOBIN=//dist
+ mkdir -p //dist
+ go_mod_install github.com/gogo/protobuf/protoc-gen-gogofast
++ awk '{print $1}'
++ go list -f '{{.Module}}' github.com/gogo/protobuf/protoc-gen-gogofast
no required module provides package github.com/gogo/protobuf/protoc-gen-gogofast: go.mod file not found in current directory or any parent directory; see 'go help modules'
+ module=
```

Reply:
Ah yes, I forgot to update Dockerfile to also add go.mod and go.sum. I'm testing this now.